flows/contact.go

```diff
@@ -360,6 +382,10 @@ func (c *Contact) ReevaluateDynamicGroups(env envs.Environment) ([]*Group, []*Gr
 	removed := make([]*Group, 0)
 	errs := make([]error, 0)
 
+	if c.Blocked() || c.Stopped() {
+		return added, removed, errs
```

Comment:
isn't this supposed to remove the contact from groups?

Reply:
i.e. return all groups as in removed?

Reply:
I think this is a protection to make sure that we can not recalculate the groups for blocked contacts, 
and the groups are removed on https://github.com/nyaruka/goflow/pull/902/files#diff-1f0f8102df6183de84f3f896458164e3R43 and their events are produced there

Reply:
This seems more complicated than it needs to be... if `Contact.ReevaluateDynamicGroups` knows that blocked/stopped means remove from any group, then all the modifiers have to do is

1. Change the state of `stopped` / `blocked`
2. Call `m.reevaluateDynamicGroups(env, assets, contact, log)`

Wouldn't that work?

Reply:
Yep, that will work